frontend/src/components/eMIB/EditActionDialog.jsx

```diff
@@ -108,19 +113,113 @@ class EditActionDialog extends Component {
     this.setState({ action: updatedAction });
   };
 
-  showCancelConfirmationDialog = () => {
-    this.setState({ showCancelConfirmationDialog: true });
+  handleCancelConfirmationDisplay = () => {
+    // ======================================== VARIABLES ========================================
+
+    // setting initial email and task forms variables to empty strings
+    let {
+      initialEmailType,
+      initialEmailTo,
+      initialEmailCc,
+      initialEmailResponse,
+      initialTaskContent,
+      initialReasonsForActionContent
+    } = "";
+
+    // get current email form variables
+    const emailType = this.state.action.emailType;
+    const emailTo = this.state.action.emailTo;
+    const emailCc = this.state.action.emailCc;
+    const emailResponse = this.state.action.emailBody;
+
+    // get current task form variables
+    const taskContent = this.state.action.task;
+    const reasonsForActionContent = this.state.action.reasonsForAction;
+    // =============================================================================================
+
+    // no content has been added in the concerned form (Email Response Form or Task Form)
```

Comment:
I would also mention that this logic is also for the creation of a new task/email. Something like "(New Email Response Form....)"

Reply:
though you can have a new email in the other part... this first part is just for a new email with an empty form

Reply:
Added comments in commit #https://github.com/code-for-canada/project-thundercat/pull/229/commits/2ffba3713cadbb590f4fcc512a45eaaa15c24c65